src/Provider/AbstractProvider.php

```diff
@@ -638,10 +638,10 @@ protected function getDefaultHeaders($token = null)
      *
      * No default is provided, providers must overload this method to activate
      * authorization headers.
-     *
+     * @param AccessToken|null $token
      * @return array
      */
-    protected function getAuthorizationHeaders($token = null)
+    protected function getAuthorizationHeaders(AccessToken $token = null)
```

Comment:
I don't agree with this change. One of the benefits of the `AccessToken` is that it includes a `__toString` method, which means that we don't need to care about whether we only store the token as a string, or recreate a full instance of the access token.